### ISSUE TITLE
Issue creating (init) new site with thor 1.3.0

### ISSUE THREAD

**Issue by ttych** [middleman/middleman#2665]
## Expected behavior and actual behavior

From a fresh install of middleman,
I am not able to create a new site with **init** command.
There is an error between Thor and middleman.

I put all the details in the following parts.
I did not get what changed between 1.2.2 and 1.3.0 in Thor to explain this error now.

**Many thanks !**

## Steps to reproduce the problem (from a clean middleman installation)

From a fresh ruby install (with rbenv), install middleman

``` shell
$ gem install middleman
Fetching execjs-2.9.1.gem                                                      
Fetching servolux-0.13.0.gem                                                   
Fetching uglifier-3.2.0.gem                                                    
Fetching tilt-2.0.11.gem                                                       
Fetching toml-0.3.0.gem                                                        
Fetching webrick-1.8.1.gem                                                     
Fetching ffi-1.16.3.gem                                                                                                                                        
Fetching parslet-2.0.0.gem                                                     
Fetching sassc-2.4.0.gem                                                       
Fetching rack-2.2.8.gem                                                        
Fetching parallel-1.23.0.gem                                                   
Fetching padrino-support-0.15.3.gem                                            
Fetching concurrent-ruby-1.2.2.gem                                             
Fetching i18n-1.6.0.gem                                                        
Fetching padrino-helpers-0.15.3.gem                                            
Fetching memoist-0.16.2.gem                                                    
Fetching rb-inotify-0.10.1.gem                                                 
Fetching rb-fsevent-0.11.2.gem                                                 
Fetching listen-3.8.0.gem                                                      
Fetching hashie-3.6.0.gem                                                      
Fetching hamster-3.0.0.gem                                                     
Fetching fastimage-2.2.7.gem                                                   
Fetching fast_blank-1.0.1.gem                                                  
Fetching erubis-2.7.0.gem                                                      
Fetching dotenv-2.8.1.gem                                                      
Fetching contracts-0.17.gem                                                    
Fetching backports-3.24.1.gem                                                  
Fetching public_suffix-5.0.3.gem                                               
Fetching addressable-2.8.5.gem                                                 
Fetching tzinfo-2.0.6.gem                                                      
Fetching activesupport-7.0.8.gem                                               
Fetching middleman-core-4.5.0.gem                                              
Fetching thor-1.3.0.gem                                                        
Fetching middleman-cli-4.5.0.gem                                               
Fetching kramdown-2.4.0.gem                                                    
Fetching temple-0.10.3.gem                                                     
Fetching minitest-5.20.0.gem                                                   
Fetching haml-6.2.3.gem                                                        
Fetching coffee-script-source-1.12.2.gem
Fetching coffee-script-2.4.1.gem                                               
Fetching middleman-4.5.0.gem                                                   
Fetching rexml-3.2.6.gem                                                       
Successfully installed webrick-1.8.1                                           
Successfully installed execjs-2.9.1                                            
Successfully installed uglifier-3.2.0                                          
Successfully installed parslet-2.0.0                                           
Successfully installed toml-0.3.0                                              
Successfully installed tilt-2.0.11                                             
Successfully installed servolux-0.13.0                                         
Building native extensions. This could take a while...
Successfully installed ffi-1.16.3                                              
Building native extensions. This could take a while...
Successfully installed sassc-2.4.0
Successfully installed rack-2.2.8
Successfully installed parallel-1.23.0
Successfully installed padrino-support-0.15.3
Successfully installed concurrent-ruby-1.2.2

HEADS UP! i18n 1.1 changed fallbacks to exclude default locale.
But that may break your application.

Please check your Rails app for 'config.i18n.fallbacks = true'.
If you're using I18n (>= 1.1.0) and Rails (< 5.2.2), this should be
'config.i18n.fallbacks = [I18n.default_locale]'.
If not, fallbacks will be broken in your app by I18n 1.1.x.

For more info see:
https://github.com/svenfuchs/i18n/releases/tag/v1.1.0

Successfully installed i18n-1.6.0
Successfully installed padrino-helpers-0.15.3
Successfully installed memoist-0.16.2
Successfully installed rb-inotify-0.10.1
Successfully installed rb-fsevent-0.11.2
Successfully installed listen-3.8.0
Successfully installed hashie-3.6.0
Successfully installed hamster-3.0.0
Successfully installed fastimage-2.2.7
Building native extensions. This could take a while...
Successfully installed fast_blank-1.0.1
Successfully installed erubis-2.7.0
Successfully installed dotenv-2.8.1

    0.16.x will be the supporting Ruby 2.x and be feature frozen (only fixes will be released)
    For Ruby 3.x use 0.17.x or later (might not be released yet)
   
Successfully installed contracts-0.17
Successfully installed backports-3.24.1
Successfully installed public_suffix-5.0.3
Successfully installed addressable-2.8.5
Successfully installed tzinfo-2.0.6
Successfully installed minitest-5.20.0
Successfully installed activesupport-7.0.8
Successfully installed middleman-core-4.5.0
Successfully installed thor-1.3.0
Successfully installed middleman-cli-4.5.0
Successfully installed rexml-3.2.6
Successfully installed kramdown-2.4.0
Successfully installed temple-0.10.3
Successfully installed haml-6.2.3
Successfully installed coffee-script-source-1.12.2
Successfully installed coffee-script-2.4.1
Successfully installed middleman-4.5.0
42 gems installed
```

init site : 
``` shell
$ middleman init site
         run  git clone --depth 1 https://github.com/middleman/middleman-templates-default.git /tmp/d20231019-237543-1ry8k from "."
Cloning into '/tmp/d20231019-237543-1ry8k'...
remote: Enumerating objects: 19, done.
remote: Counting objects: 100% (19/19), done.
remote: Compressing objects: 100% (11/11), done.
remote: Total 19 (delta 0), reused 13 (delta 0), pack-reused 0
Receiving objects: 100% (19/19), done.
<home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor/invocation.rb:112:in `invoke': Missing Thor class for invoke middleman:generator (RuntimeError)
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/middleman-cli-4.5.0/lib/middleman-cli/init.rb:77:in `block in init'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor/actions.rb:190:in `block in inside'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/3.2.0/fileutils.rb:244:in `chdir'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/3.2.0/fileutils.rb:244:in `cd'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor/actions.rb:190:in `inside'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/middleman-cli-4.5.0/lib/middleman-cli/init.rb:71:in `init'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor/command.rb:28:in `run'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor/invocation.rb:127:in `invoke_command'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor/invocation.rb:134:in `block in invoke_all'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor/invocation.rb:134:in `each'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor/invocation.rb:134:in `map'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor/invocation.rb:134:in `invoke_all'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor/group.rb:232:in `dispatch'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor/invocation.rb:116:in `invoke'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor.rb:40:in `block in register'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor/command.rb:28:in `run'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor/invocation.rb:127:in `invoke_command'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor.rb:527:in `dispatch'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/thor-1.3.0/lib/thor/base.rb:584:in `start'
        from <home>/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/middleman-cli-4.5.0/bin/middleman:70:in `<top (required)>'
        from <home>/.rbenv/versions/3.2.2/bin/middleman:25:in `load'
        from <home>/.rbenv/versions/3.2.2/bin/middleman:25:in `<main>'
1
```

Then fallback to thor 1.2.2 : 

``` shell
$ gem install thor -v 1.2.2
Fetching thor-1.2.2.gem
Successfully installed thor-1.2.2
1 gem installed
```

Then uninstall thor 1.3.0 : 
``` shell
$ gem uninstall thor -v 1.3.0
Successfully uninstalled thor-1.3.0
```

Then retry init:
```
$ middleman init site
         run  git clone --depth 1 https://github.com/middleman/middleman-templates-default.git /tmp/d20231019-240447-mxklq4 from "."
Cloning into '/tmp/d20231019-240447-mxklq4'...
remote: Enumerating objects: 19, done.
remote: Counting objects: 100% (19/19), done.
remote: Compressing objects: 100% (11/11), done.
remote: Total 19 (delta 0), reused 13 (delta 0), pack-reused 0
Receiving objects: 100% (19/19), done.
       exist  
   identical  .gitignore
   identical  Gemfile
   identical  config.rb
   identical  source/images/.keep
   identical  source/index.html.erb
   identical  source/javascripts/site.js
   identical  source/layouts/layout.erb
   identical  source/stylesheets/site.css.scss
         run  bundle install from "./site"
Fetching gem metadata from https://rubygems.org/............
Using backports 3.24.1
Using coffee-script-source 1.12.2
Using concurrent-ruby 1.2.2
Using erubis 2.7.0
Using public_suffix 5.0.3
Using execjs 2.9.1
Using ffi 1.16.3
Using bundler 2.4.10
Using dotenv 2.8.1
Using minitest 5.20.0
Using hashie 3.6.0
Using fast_blank 1.0.1
Using contracts 0.17
Using temple 0.10.3
Fetching thor 1.3.0
Using parallel 1.23.0
Using rack 2.2.8
Using servolux 0.13.0
Using parslet 2.0.0
Using webrick 1.8.1
Using i18n 1.6.0
Using tzinfo 2.0.6
Using tilt 2.0.11
Using rb-fsevent 0.11.2
Using rexml 3.2.6
Fetching autoprefixer-rails 9.8.6.5
Using fastimage 2.2.7
Using coffee-script 2.4.1
Using memoist 0.16.2
Using padrino-support 0.15.3
Using sassc 2.4.0
Using hamster 3.0.0
Using rb-inotify 0.10.1
Using addressable 2.8.5
Using uglifier 3.2.0
Using activesupport 7.0.8
Using toml 0.3.0
Using kramdown 2.4.0
Using padrino-helpers 0.15.3
Using listen 3.8.0
Using middleman-core 4.5.0
Installing thor 1.3.0
Installing autoprefixer-rails 9.8.6.5
Using haml 6.2.3
Using middleman-cli 4.5.0
Using middleman 4.5.0
Fetching middleman-autoprefixer 2.10.1
Installing middleman-autoprefixer 2.10.1
Bundle complete! 4 Gemfile dependencies, 45 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
```

## Additional information

- Ruby version: 3.2.2
- Middleman version: 4.5.0
- OS version: Ubuntu 23.04

**Comment by MattBudz** [comment]
I can also replicate this on:
- Ruby 3.2.2
- Middleman 4.5.0
- macOS 13.1 and 14.0

**Comment by tdreyno** [comment]
Locked Thor dep to 1.2.x and published new middleman version 4.5.1
